### PR TITLE
Fixed testing errors

### PR DESCRIPTION
**Pull request by blake-gilmore**
## Issue Number
Addresses part of https://github.com/UNLV-CS472-672/2024-S-GROUP4-SMTK/issues/58 Expand Test Coverage to 90% but does not close it.

## Description
This PR fixes the errors that were occurring when running npm run test:coverage with jest. Before, jest was having issues with rendering the component likely due to how the SideBarButtons were being mocked. I did a couple things to improve this test.

- Removed the const buttons list from the SideBar component and added it to its own app/data/sidebarButtons.json. This will make it much easier for testing, so we can simply fetch how the buttons are supposed to look and compare to how the test is rendered. This way, if anything is changed with the json information, the test will still pass, because we're not exactly testing for the content of the buttons, but that the SideBar component is correctly rendering the buttons from its json.
- Created a mocks folder in components and made a SidebarButton.js mock file for testing. I was getting errors from jest with how the mock was being created, and it seemed that the mock wasn't being properly rendered. I referenced this stackoverflow for my fix: https://stackoverflow.com/questions/44403165/using-jest-to-mock-a-react-component-with-props Instead of mocking the button within the test file, it now calls a mock for the SidebarButton, and jest will automatically find the mock in the mocks folder.

## Testing
The tests on SideBar.test.js now pass correctly

## Possible Errors
It's possible that the change in the test here go against what the tests were originally made to check for. This would be related to @AlvaroH2001 PR https://github.com/UNLV-CS472-672/2024-S-GROUP4-SMTK/pull/66

### CODE REVIEW

**Review by AlvaroH2001 — APPROVED** [review]
Excellent additions and structural changes, this form of thinking and problem solving will certainly help in future tests and separating components for easier testing.

File: small-talk/app/components/SideBar.js [hunk 1]
```diff
@@ -1,18 +1,10 @@
 import React from 'react';
 import SidebarButton from "./SidebarButton";
+import buttons from "../data/sidebarButtons.json"
```
Comment (junxjr) [inline small-talk/app/components/SideBar.js:3]: Looks good, makes the file really clean.

**Review by 39otsu — APPROVED** [review]
This pull request also marks a good example on how to test the rendering of components using jest. Great addition!

**Review by junxjr — APPROVED** [review]
I have ran the preview version and the code, everything looks good to me. By having its own file for sidebar buttons makes it easier for the future, and it is really clean.

**Pull request merged**